demos/interactive_face_detection_demo_gapi/main.cpp

```diff
@@ -0,0 +1,556 @@
+    for (size_t i = 0; i < emotionsVecSize; i++) {
+        std::cout << emotionsVec[i] << " = " << em_data[i];
+        if (emotionsVecSize - 1 != i) {
+            std::cout << ", ";
+        } else {
+            std::cout << std::endl;
+        }
+    }
+}
+
+void faceDataUpdate(const cv::Mat &frame,
+                    Face::Ptr &face,
+                    const cv::Rect &face_rect,
+                    std::list<Face::Ptr>  &prev_faces,
+                    const std::vector<cv::Rect> &face_hub,
+                    size_t &id,
+                    bool no_smooth) {
+    // Face update
+    cv::Rect rect = face_rect & cv::Rect({0, 0}, frame.size());
+
+    if (!no_smooth) {
+        face = matchFace(rect, prev_faces);
+        float intensity_mean = calcMean(frame(rect));
+        intensity_mean += 1.0;
+
+        if ((face == nullptr) ||
+            ((face != nullptr) && ((std::abs(intensity_mean - face->_intensity_mean) / face->_intensity_mean) > 0.07f))) {
+            face = std::make_shared<Face>(id++, rect);
+        } else {
+            prev_faces.remove(face);
+        }
+
+        face->_intensity_mean = intensity_mean;
+        face->_location = rect;
+    } else {
+        face = std::make_shared<Face>(id++, rect);
+    }
+}
+
+void ageGenderDataUpdate(const Face::Ptr &face,
+                         const cv::Mat &out_age,
+                         const cv::Mat &out_gender) {
+    const float *age_data =    out_age.ptr<float>();
+    const float *gender_data = out_gender.ptr<float>();
+
+    float maleProb = gender_data[1];
+    float age      = age_data[0] * 100;
+
+    face->updateGender(maleProb);
+    face->updateAge(age);
+}
+
+void headPoseDataUpdate(const Face::Ptr &face,
+                        const cv::Mat &out_y_fc,
+                        const cv::Mat &out_p_fc,
+                        const cv::Mat &out_r_fc) {
+    const float *y_data = out_y_fc.ptr<float>();
+    const float *p_data = out_p_fc.ptr<float>();
+    const float *r_data = out_r_fc.ptr<float>();
+
+    face->updateHeadPose(y_data[0], p_data[0], r_data[0]);
+}
+
+void emotionsDataUpdate(const Face::Ptr &face, const cv::Mat &out_emotion) {
+    const float *em_data = out_emotion.ptr<float>();
+
+    face->updateEmotions({
+                          {"neutral", em_data[0]},
+                          {"happy", em_data[1]} ,
+                          {"sad", em_data[2]} ,
+                          {"surprise", em_data[3]},
+                          {"anger", em_data[4]}
+                          });
+}
+
+void landmarksDataUpdate(const Face::Ptr &face, const cv::Mat &out_landmark) {
+    const float *lm_data = out_landmark.ptr<float>();
+
+    std::vector<float> normedLandmarks;
+    size_t n_lm = 70;
+    for (size_t i_lm = 0UL; i_lm < n_lm / 2; ++i_lm) {
+        normedLandmarks.push_back(lm_data[2 * i_lm]);
+        normedLandmarks.push_back(lm_data[2 * i_lm + 1]);
+    }
```

Comment:
Doesn't this basically just copy `lm_data` to `normedLandmarks`?

Reply:
Since I receive the data in the form of a `cv::Mat`, it is easier to first copy it into a `vector`, so that later it can be conveniently used

Used `normedLandmarks.insert(normedLandmarks.end(), &lm_data[0], &lm_data[n_lm]);`